### PR TITLE
Hints on deposit and withdraw

### PR DESCRIPTION
**Pull request by brittcyr**

### CODE REVIEW

File: programs/ui-wrapper/src/processors/place_order.rs [hunk 1]
```diff
@@ -156,6 +156,7 @@ pub(crate) fn process_place_order(
                 deposit_amount_atoms,
                 trader_token_account.key,
                 *token_program.key,
+                None,
```
Comment (mschneider) [inline programs/ui-wrapper/src/processors/place_order.rs:159]: Some(trader_index)

File: programs/ui-wrapper/src/processors/settle_funds.rs [hunk 2]
```diff
@@ -123,6 +123,7 @@ pub(crate) fn process_settle_funds(
             base_balance.as_u64(),
             trader_token_account_base.key,
             *token_program_base.key,
+            None,
```
Comment (mschneider) [inline programs/ui-wrapper/src/processors/settle_funds.rs:126]: could look up Some(trader_index)

File: programs/ui-wrapper/src/processors/settle_funds.rs [hunk 3]
```diff
@@ -144,6 +145,7 @@ pub(crate) fn process_settle_funds(
             quote_balance.as_u64(),
             trader_token_account_quote.key,
             *token_program_quote.key,
+            None,
```
Comment (mschneider) [inline programs/ui-wrapper/src/processors/settle_funds.rs:148]: could use Some(trader_index)

File: programs/manifest/src/program/processor/shared.rs [hunk 4]
```diff
@@ -157,3 +166,48 @@ pub fn get_dynamic_value<T: Get>(data: &[u8]) -> DynamicAccount<T, Vec<u8>> {
...
+
+fn verify_trader_index_hint(
+    hinted_index: DataIndex,
+    dynamic_account: &MarketRefMut,
+    payer: &Signer,
+) -> ProgramResult {
+    require!(
+        hinted_index % (MARKET_BLOCK_SIZE as DataIndex) == 0,
+        ManifestError::WrongIndexHintParams,
+        "Invalid trader hint index {}",
```
Comment (mschneider) [inline programs/manifest/src/program/processor/shared.rs:194]: would favor more descriptive messages here:
- hint index not aligned
- expected claimed seat at hint index 
- expected claimed seat at hint index to be owned by payer

Comment (brittcyr) [reply programs/manifest/src/program/processor/shared.rs]: Just the message, or need a different error too?

**Pull request merged**